utils/getCiData.js

```diff
@@ -0,0 +1,50 @@
+const { execToStr } = require('./execTo');
+
+const getCiData = () => {  
+  if (process.env.NETLIFY) {
```

Comment:
Rather than doing multiple `if` / `else if` statements, we might want to consider using a `switch` or a lookup object.

Furthermore, if only one of these CI environments can be active set at a time, we should probably normalize the data into something like `process.env.CI_ENVIRONMENT`).

If we did that, then it would make a `switch` really straightforward:

```js
const getCIData = () => {
  switch (process.env.CI_ENVIRONMENT) {
    case: "netlify":
      return { /* netlify-specific configuration */ };
    case: "vercel":
      return { /* ... */ };
    // etc...
    default:
      return { /* default configuration */ };
  }
}
```

Same with a lookup object:

```js
const CI_CONFIG = {
  netlify: { /* netlify-specific configuration */ },
  vercel: { /* ... */ },
  // etc..
  default: { /* defailt configuration */ }
};

const getCIData = () => CI_CONFIG[process.env.CI_ENVIRONMENT || "default"];
```

Reply:
at the moment the CI specific environment variables like `process.env.NETLIFY` are either set to `true` if the build is happening on that CI or undefined, so i'm not sure if those options would work in that case. Though there may also be other env's that would be more appropriate to leverage. The `CI_NAME` is one that comes to mind, but will have to do testing on each platform to make sure that would work

Reply:
Ah interesting. That's a very weird pattern, because we need to add a _new_ variable for each environment. Furthermore, it doesn't look like the code supports more than one environment (`if else` not `if`).

If we don't have a consistent variable, but we still wanted to use a `switch`, there's a trick for that:

```js
const getCIData = () => {
  switch (true) {
    case process.env.NETLIFY:
      return { /* ... */ };
    case process.env.VERCEL:
      return { /* ... */ };
    // etc...
    default:
      return { /* ... */ };
  }
};
```

Reply:
sweet this works nicely :D